docs/zkapps/tutorials/10-account-updates.mdx

```diff
@@ -23,166 +23,187 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
+In this tutorial, you build two smart contracts: 
 
-This zkApp also has methods that call other methods, so you can explore how that impacts a transaction's AccountUpdates.
+- `ProofsOnlyZkApp`: Non-upgradeable proof only
+- `SecondaryZkApp`: the other zkApp
 
-Start by adding the main contents of the zkApp:
+The full source code for this tutorial is provided in the [examples/zkapps/10-account-updates](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/10-account-updates) directory on GitHub.
 
-```typescript
-export class ProofsOnlyZkApp extends SmartContract {
-  @state(Field) num = State<Field>();
-  @state(Field) calls = State<Field>();
-
-  deploy(args: DeployArgs) {
-    super.deploy(args);
-    this.setPermissions({
-      ...Permissions.default(),
-      setDelegate: Permissions.proof(),
-      setPermissions: Permissions.proof(),
-      setVerificationKey: Permissions.proof(),
-      setZkappUri: Permissions.proof(),
-      setTokenSymbol: Permissions.proof(),
-      incrementNonce: Permissions.proof(),
-      setVotingFor: Permissions.proof(),
-      setTiming: Permissions.proof(),
-    });
-  }
+### Non-upgradeable proof only
 
-  @method init() {
-    this.account.provedState.assertEquals(this.account.provedState.get());
-    this.account.provedState.get().assertFalse();
+The full example code is provided in the [examples/zkapps/10-account-updates/src/ProofsOnlyZkApp.ts](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/10-account-updates/src/ProofsOnlyZkApp.ts) file.
 
-    super.init();
-    this.num.set(Field(1));
-    this.calls.set(Field(0));
-  }
+Goal: Configure this zkApp to be modifiable only by using proofs. 
 
-  ...
-```
+For this example, the zkApp is not upgradable after it is deployed. This means that while the zkApp developer owns the private key to initially deploy the zkApp, after its first deployment, the zkApp requires proof authorization and consequently can be updated only transactions that fulfill the zkApp smart contract logic. The private key is no longer useful for anything.
 
-This code configures the zkApp as described and initializes the zkApp with the values you want.
+This zkApp has methods that call other methods to let you explore the impacts to a transaction's account updates.
 
-:::caution
-Note you assert that `provedState` is `false` in `init()` to ensure that `init()` cannot be called again after the zkApp is set up during the initial deployment. Without this, your zkApp could be reset by anyone calling the `init()` method on your zkApp. This is recommended for most zkApps.
-:::
+1. Start by adding the main contents of the zkApp:
 
-Next, add two functions:
+  ```typescript
+  export class ProofsOnlyZkApp extends SmartContract {
+    @state(Field) num = State<Field>();
+    @state(Field) calls = State<Field>();
 
-```typescript
-  ...
+    deploy(args: DeployArgs) {
```

Comment:
These code should eventually be updated for the most recent o1js, but I won't have time to do it today :(
Would it be okay to merge this and put a card on the board to update this code in the next sprint?

Reply:
we'll track this work in https://github.com/o1-labs/docs2/issues/660